src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx

```diff
@@ -80,6 +94,33 @@ export const MyCollectionArtworkForm: React.FC<MyCollectionArtworkFormProps> = (
         artworkLocation: values.artworkLocation,
       })
 
+      // Remove photos marked for deletion
+
+      const removedPhotos = values.photos.filter(photo => photo.removed)
+
+      await Promise.all(
+        removedPhotos.map(async photo => {
+          try {
+            await deleteArtworkImage({
```

Comment:
Oh, that's a tricky one! The code should remove the photos in parallel because `map()` is not awaiting its callback function and therefore running the function for all elements in parallel. [here](https://codesandbox.io/s/javascript-forked-x3ehzq?file=/index.js) is a simplified example.